anndata/_core/anndata.py

```diff
@@ -1268,7 +1273,17 @@ def transpose(self) -> "AnnData":
             )
 
         def t_csr(m: sparse.spmatrix) -> sparse.csr_matrix:
-            return m.T.tocsr() if sparse.isspmatrix_csr(m) else m.T
+            if m is None:
+                return None
+            elif sparse.isspmatrix_csr(m):
+                return m.T.tocsr()
+            else:
+                return m.T
+
+        if X is not None:
+            dtype = X.dtype
+        else:
+            dtype = "float32"
 
         return AnnData(
             t_csr(X),
```

Comment:
Instead adding all that stuff to `t_csr`, could this be: `t_csr(X) if X is not None else None`

My thinking here is that I'm not sure if we should keep the behavior of `t_csr`, so I don't want to put more logic in it.

Reply:
Fair enough!